system-addon/test/unit/lib/PrefsFeed.test.js

```diff
@@ -41,4 +45,38 @@ describe("PrefsFeed", () => {
     feed.onPrefChanged("foo", 2);
     assert.calledWith(feed.store.dispatch, ac.BroadcastToContent({type: at.PREF_CHANGED, data: {name: "foo", value: 2}}));
   });
+  describe("INIT prerendering", () => {
+    it("should set a prerender pref on init", () => {
+      feed.onAction({type: at.INIT});
+      assert.calledWith(feed._prefs.set, PRERENDER_PREF_NAME);
+    });
+    it("should set prerender pref to true if prefs match initial values", () => {
+      Object.keys(initialPrefs).forEach(name => FAKE_PREFS.set(name, initialPrefs[name]));
+      feed.onAction({type: at.INIT});
+      assert.calledWith(feed._prefs.set, PRERENDER_PREF_NAME, true);
+    });
+    it("should set prerender pref to false if a pref does not match its initial value", () => {
+      Object.keys(initialPrefs).forEach(name => FAKE_PREFS.set(name, initialPrefs[name]));
+      FAKE_PREFS.set("feeds.section.topstories", false);
+      feed.onAction({type: at.INIT});
+      assert.calledWith(feed._prefs.set, PRERENDER_PREF_NAME, false);
+    });
+  });
+  describe("onPrefChanged prerendering", () => {
+    it("should not change the prerender pref if the pref is not included in PrerenderData.invalidatingPrefs", () => {
+      feed.onPrefChanged("foo123", true);
+      assert.notCalled(feed._prefs.set);
+    });
+    it("should set the prerender pref if one of the prefs in PrerenderData.invalidatingPrefs is changed", () => {
```

Comment:
This wants to be split into two tests for the true -> false change and the false -> true change.